src/libs/error.js

```diff
@@ -15,29 +15,28 @@ export const reportError = async (title, obj) => {
   }
 }
 
+/**
+ * Invoke the `callback` with any error thrown when evaluating the async `fn` with `...args`.
+ */
+export const withErrorHandling = _.curry((callback, fn) => async (...args) => {
+  try { return await fn(...args) } catch (error) { await callback(error) }
+})
+
 /**
  * Return a Promise to the result of evaluating the async `fn` with `...args` or undefined if
  * evaluation fails.
  */
-export const withErrorIgnoring = fn => async (...args) => {
-  try {
-    return await fn(...args)
-  } catch (error) {
-    // ignore error
-  }
-}
+export const withErrorIgnoring = withErrorHandling(() => {})
 
 /**
  * Return a Promise to the result of evaluating the async `fn` with `...args`. If evaluation fails,
  * report the error to the user with `title` as a side effect.
  */
-export const reportErrorAndRethrow = _.curry((title, fn) => async (...args) => {
-  try {
-    return await fn(...args)
-  } catch (error) {
+export const reportErrorAndRethrow = _.curry((title, fn) => {
+  return _.flip(withErrorHandling)(fn, error => {
```

Comment:
Is `_.flip` needed here?

Reply:
Needed? No. Then again, I think it's nicer to look at than: 
```javascript
return withErrorHandling(error => {
    reportError(title, error)
    throw error
 }, fn)
```